File: server/models/post.js
const mongoose = require("mongoose")

const PostSchema = new mongoose.Schema({
  authorId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User",
    required: true,
  },
  content: {
    type: String,
  },
  imageUrl: {
    type: String,
  },
  likes: [
    {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
    },
  ],
  comments: [
    {
      user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "User",
      },
      text: {
        type: String,
        required: true,
      },
      date: {
        type: Date,
        default: Date.now,
      },
    },
  ],
  savedBy: [
    {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
    },
  ],
  createdAt: {
    type: Date,
    default: Date.now,
  },
})

module.exports = mongoose.model("Post", PostSchema)

File: lib/models/post.dart
class Post {
  final String id;
  final String authorId;
  final String authorName;
  final String? authorImage;
  final String? content;
  final String? imageUrl;
  final int likes;
  final int comments;
  final String timeAgo;
  final bool isLiked;
  final bool isSaved;

  Post({
    required this.id,
    required this.authorId,
    required this.authorName,
    this.authorImage,
    this.content,
    this.imageUrl,
    required this.likes,
    required this.comments,
    required this.timeAgo,
    this.isLiked = false,
    this.isSaved = false,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'],
      authorId: json['authorId'],
      authorName: json['authorName'],
      authorImage: json['authorImage'],
      content: json['content'],
      imageUrl: json['imageUrl'],
      likes: json['likes'],
      comments: json['comments'],
      timeAgo: json['timeAgo'],
      isLiked: json['isLiked'] ?? false,
      isSaved: json['isSaved'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'authorId': authorId,
      'authorName': authorName,
      'authorImage': authorImage,
      'content': content,
      'imageUrl': imageUrl,
      'likes': likes,
      'comments': comments,
      'timeAgo': timeAgo,
      'isLiked': isLiked,
      'isSaved': isSaved,
    };
  }
}

